### PR TITLE
CR-1070609 XRT crashes when subdev index is out of range 0-63

### PR DESCRIPTION
**Pull request by xdavidz**
boundary checking

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c [hunk 1]
```diff
@@ -2131,6 +2131,12 @@ static void icap_calib(struct icap *icap, bool retain)
 		if (!mem_topo->m_mem_data[i].m_used)
 			continue;
 
+		if (ddr_idx >= XOCL_SUBDEV_MAX_INST) {
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c:2134]: is this a coverity defects or you did see a crash with ddr index > 64?

**Comment by xdavidz** [comment]
There will be a crash, page fault, beyond 64.

/David

On Jul 28, 2020, at 8:34 PM, Max Zhen <notifications@github.com> wrote:

﻿
CAUTION: This message has originated from an External Source. Please use proper judgment and caution when opening attachments, clicking links, or responding to this email.


@maxzhen commented on this pull request.

________________________________

In src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c<https://github.com/Xilinx/XRT/pull/3766#discussion_r462017707>:

> @@ -2131,6 +2131,12 @@ static void icap_calib(struct icap *icap, bool retain)
                if (!mem_topo->m_mem_data[i].m_used)
                        continue;

+               if (ddr_idx >= XOCL_SUBDEV_MAX_INST) {


is this a coverity defects or you did see a crash with ddr index > 64?

**Comment by xdavidz** [comment]
Chien-wei thinks we should handle this with in DEV_MULTI macro, but as of today it is not a one line fix. 
Will improve the APIs in future fix.

You may close this PR as "will-not-fix". I will put a reason in the CR too.

**Comment by maxzhen** [comment]
Closing per request from @xdavidz

**Pull request closed without merging**